src/ispc.h

```diff
@@ -146,6 +146,14 @@ struct SourcePos {
     bool operator==(const SourcePos &p2) const;
 };
 
+/** Structure to represent float16 constant value. */
+struct float16Ty {
+    float16Ty() : isStr(true) {}
+    float fval;
```

Comment:
I'm wondering if we can use `APFloat` class for storing `float16` value right aways, without go through `float` type.

Reply:
I'm not storing it as float when the parser identifies a fp16 constant. In these cases I directly use this str to create a  APFloat. The reason for the option to store as float is there are othe ways bu which an fp16 is created

1. Reading fp16 directly from parser/lexer(*fp16 suffix) - this case we simple store it as  str and create APFloat directly from this
2. Parser read a float/double/int etc and it is stored as suct in AST. This si then typecast to fp16. In theses cases, tehe existing float/double/int constant is stored as such. I cannot create a APFloat from string in these cases. The option I went with was store these as float and use APFloat to convert it to falf

Reply:
Done